packages/test/test-end-to-end-tests/src/test/containerDirtyFlag.spec.ts

```diff
@@ -35,16 +35,20 @@ const testValue = "test value";
 
 type MapCallback = (container: IContainer, dataStore: ITestFluidObject, map: SharedMap) => void | Promise<void>;
 
-// load container, pause, create (local) ops from callback, then optionally send ops before closing container
-const getPendingOps = async (args: ITestObjectProvider, send: boolean, cb: MapCallback) => {
-    const container = await args.loadTestContainer(testContainerConfig);
+const resolveConnection = async(container: IContainer) => {
     await new Promise<void>((resolve) => {
         if ((container as any).connected) {
             resolve();
         } else {
             container.on("connected", () => resolve());
```

Comment:
If you search for `ensureContainerConnected` and `waitForContainerReconnection` functions, you will see that we wait for container to get connected in a lot of tests. I have seen that if we don't do this, sometimes `ensureSynchronized` hangs. I wonder if there is an underlying issue with `ensureSynchronized` here.

Reply:
Yeah - underlining issue may be around there. We should probably not be exiting out ensureSynchronized having container not connected. Let's dig into this as part of that other ticket above.